src/Sylius/Bundle/ApiBundle/Validator/Constraints/OrderProductEligibilityValidator.php

```diff
@@ -46,6 +47,10 @@ public function validate(mixed $value, Constraint $constraint): void
         /** @var OrderItemInterface[] $orderItems */
         $orderItems = $order->getItems();
 
+        /** @var ChannelInterface|null $channel */
+        $channel = $order->getChannel();
+        Assert::notNull($channel);
```

Comment:
It's not the responsibility of this validator to check the domain preconditions.
```suggestion
        if ($channel === null) {
            return;
        }
```

Reply:
👍🏻